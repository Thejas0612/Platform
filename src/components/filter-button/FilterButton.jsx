import React, { useState } from "react";
import PropTypes from "prop-types";
import { Button } from "@mui/material";
import TuneIcon from "@mui/icons-material/Tune";
import CloseIcon from "@mui/icons-material/Close";
import { MsolComponentHighlighter } from "../msol-component-highlighter/MsolComponentHighlighter";
import { grey } from "@mui/material/colors";

export const FilterButton = ({
                        label,
                        onClick,
                      }) => {
  const [expanded, setExpanded] = useState(false);

  const handleButtonClick = () => {
    setExpanded(!expanded);
    onClick();
  };

  return (
    <MsolComponentHighlighter>
      <Button
        variant="text"
        size="large"
        endIcon={expanded ? <CloseIcon /> : <TuneIcon />}
        onClick={handleButtonClick}
        sx={{
          textTransform: "none",
          color: grey[900],
        }}
      >
        {label}
      </Button>
    </MsolComponentHighlighter>
  );
};

FilterButton.propTypes = {
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};

FilterButton.defaultProps = {
  onClick: () => {},
};
