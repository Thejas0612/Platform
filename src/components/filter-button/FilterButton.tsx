import React, { FunctionComponent, useState } from "react";
import { Button } from "@mui/material";
import TuneIcon from "@mui/icons-material/Tune";
import CloseIcon from "@mui/icons-material/Close";
import { MsolComponentHighlighter } from "../msol-component-highlighter/MsolComponentHighlighter";
import { grey } from "@mui/material/colors";

export interface FilterButtonProps {
  label: string;
  onClick?: () => void;
}

export const FilterButton: FunctionComponent<FilterButtonProps> = ({
                                                                     label,
                                                                     onClick = () => {
                                                                     }
                                                                   }) => {
  const [expanded, setExpanded] = useState<boolean>(false);

  const handleButtonClick = () => () => {
    setExpanded(!expanded);
    onClick();
  };

  return <MsolComponentHighlighter>
    <Button
      variant="text"
      size="large"
      endIcon={expanded ? <CloseIcon /> : <TuneIcon />}
      onClick={handleButtonClick()}
      sx={{
        textTransform: "none",
        color: grey[900]
      }}
    >
      {label}
    </Button>
  </MsolComponentHighlighter>;
};