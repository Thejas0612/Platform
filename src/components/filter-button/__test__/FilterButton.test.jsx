import { FilterButton } from "../FilterButton";
import { fireEvent, render, screen } from "@testing-library/react";

const PROPS = {
    label: "Filters"
};

describe("<TechnologyTypeFilter />", () => {
    test("should render the filter with default state", () => {
        const { container } = render(<FilterButton {...PROPS} />);

        expect(container).toMatchSnapshot();
    });

    test("should fire onClose event when clicked.", async () => {
        const handleClick = jest.fn();
        const { container } = render(<FilterButton {...PROPS} onClick={handleClick} />);

        const button = screen.getByText("Filters");
        fireEvent.click(button);

        expect(handleClick).toHaveBeenCalled();
        expect(container).toMatchSnapshot();
    });
});