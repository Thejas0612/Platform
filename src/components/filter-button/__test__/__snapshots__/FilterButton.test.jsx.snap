// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TechnologyTypeFilter /> should fire onClose event when clicked. 1`] = `
<div>
  <div
    class="MuiStack-root css-15y4ctf-MuiStack-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge css-1brjbr1-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Filters
      <span
        class="MuiButton-endIcon MuiButton-iconSizeLarge css-z4rqyd-MuiButton-endIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`<TechnologyTypeFilter /> should render the filter with default state 1`] = `
<div>
  <div
    class="MuiStack-root css-15y4ctf-MuiStack-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge css-1brjbr1-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Filters
      <span
        class="MuiButton-endIcon MuiButton-iconSizeLarge css-z4rqyd-MuiButton-endIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
